### DISCUSSION TITLE
Работать только с адресами из белого списка

### DISCUSSION THREAD

**Discussion by ilyaigpetrov** [anticensority/runet-censorship-bypass#96] [General]
# Работать только с адресами из белого списка

- Теперь можно ограничить адреса, на которых будет работать расширение. К примеру, можно указать *.kasparov.ru для работы только с этим сайтом.
- При выводе из гибернации будет делаться несколько попыток обновить PAC-скрипт. Это для случаев, когда при выходе из гибернации сеть восстанавливается с задержкой.
- Окно расширения больше не распирается в FireFox.

## Патчи

Здесь будут публиковаться патчи без уведомлений. 

<hr /><em>This discussion was created from the release <a href='https://github.com/anticensority/runet-censorship-bypass/releases/tag/0.0.1.52-rc0'>Работать только с адресами из белого списка</a>.</em>

**Comment by IgorLY** [comment]
Здравствуйте уважаемые Создатели и Разработчики Антизапрета!
Вы подготовили очень ценное дополнение  "Работать только с адресами из белого списка", но мне кажется его можно чуть упростить. Объясню почему я так думаю. Вот как представляю себе алгоритм работы:
1 При активации  Антизапрета он переходит в режим "фильтровать всё" (что вы и указали). 
2 При включении опции "Ограничиться только белым списком" он переходит в режим "фильтровать всё" если белый список пуст.
3 Примени п.2 фильтрация продолжается вне зависимости от того заполнены разделы # ПРОКСИРОВАТЬ:, # НЕ ПРОКСИРОВАТЬ: и при сохранении, а потом после загрузки введённые данные не сохраняются и не отображаются. Проверял неоднократно на хроме и хромоклонах :-).
4 А при введении адресов в "# Разрешить расширению работать только с этими адресами" всё становиться на свои места, при сохранении, а потом после загрузки введённые данные сохраняются и отображаются.  Дополнение переходит в режим "не фильтровать всё".
Таким образом разделы # ПРОКСИРОВАТЬ:, # НЕ ПРОКСИРОВАТЬ:  вообще-то не  нужны, если только они необходимы для других задач. Тогда получается следующее; активировали приложение, хочется поскорее - включили белый список  (его кстати можно хранить у пользователя, дабы не нагружать сервера), наткнулись на "мину" :-) - деактивировали белый список (пара секунд), ввели заблокированный адрес, получили адрес хоста, скопировали, открыли белый список, поместили, сохранили, активировали белый список. Всё ;-)). Возможно я ошибаюсь, но такой вариант поможет уменьшить нагрузку на ваши прокси сервера, не сразу конечно, так как люди всё-таки "лентяи" :-). Зато тем, кому захочется, смогут более гибко управлять своими подключениями.
Игорь.

**Reply by ilyaigpetrov** [reply]
> Таким образом разделы # ПРОКСИРОВАТЬ:, # НЕ ПРОКСИРОВАТЬ: вообще-то не нужны, если только они необходимы для других задач.

При активном (стоит галочка) и не пустом белом списке исключения из проксирования (ПРОКСИРОВАТЬ, НЕ ПРОКСИРОВАТЬ) не используются. Вне режима белого списка туда можно добавлять исключения из (не)проксирования. Эти исключения также можно редактировать через интерфейс: иконка синей ленты -> Исключения -> да (проксировать), нет (не проксировать), авто. Для добавления адреса в список проксируемых нужно иметь СВОИ прокси, для исключения из проксирования СВОИ прокси не нужны.

> [...]потом после загрузки введённые данные не сохраняются и не отображаются. Проверял неоднократно на хроме и хромоклонах :-).

Приведите воспроизводимый алгоритм, когда они не сохраняются. Какие шаги нужно предпринять после установки расширения, чтобы воспроизвести проблему?

> [...]включили белый список (его кстати можно хранить у пользователя, дабы не нагружать сервера[...]

Белый список хранится в памяти расширения в браузере пользователя, не на наших серверах.
